### PR TITLE
AP-4863: Add countries fuzzy search js

### PR DESCRIPTION
**Pull request by agoldstone93**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4863)

Describe what you did and why.

Update non UK home address page to conditionally show radio buttons based on what a user types in the country text field.

- Add js file and css for country search
- Add link to allow clearing search in the country field
- Rename `country` field to `country_code` and add `country_name` field to `Address` table
- Update tests to reflect changes and add cassettes for calling Legal Framework API

NB. The `country` database column has been renamed destructively as this is not yet live on production

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -113,9 +114,9 @@
     t.integer "failed_attempts", default: 0, null: false
     t.string "unlock_token"
     t.datetime "locked_at", precision: nil
-    t.boolean "employed"
     t.datetime "remember_created_at", precision: nil
     t.string "remember_token"
+    t.boolean "employed"
```
Comment (colinbruce) [inline db/schema.rb:119]: This and the change at 606 are not related to this PR.
It happens, some devs move them down and another commit moves them back it's weird and annoying, but not destructive

File: spec/factories/addresses.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
     address_line_two { Faker::Address.street_address }
     city { Faker::Address.city }
     county { Faker::Address.city }
-    country { "GBR" }
+    country_code { "GBR" }
```
Comment (colinbruce) [inline spec/factories/addresses.rb:8]: Should this add and set country_name too?

**Review by colinbruce — APPROVED** [review]
A couple of non-blocking observations, but otherwise LGTM :+1:

**Review by naseberry — APPROVED** [review]
